### ISSUE TITLE
Screenshot is incorrectly named/file is not generated in CJK language

### ISSUE THREAD

**Issue by telk5093** [OpenRCT2/OpenRCT2#7533]
**OS:** Windows 10
**Version:** 0.1.3
**Commit/Build:** 94f005a

If the scenario name is Korean(and maybe Japanese or chinese too), then screenshot file name is incorrectly named or file itself is not generated.

**Steps to reproduce:**
1. Set the language to Korean
2. Open any default scenario, such as _Roaring Twenties - Skyscrapers_ (its korean translation is ``광란의 20년대 - 마천루``)
3. Take any screenshot.
4. Game generates the screenshot as this file name: ``硫뷀듃濡쒗뤃由ъ뒪 2018-05-17 02-43-37``
5. Or game doesn't generate the screenshot file in certain scenarios.

Maybe related to #7415

**Comment by AaronVanGeffen** [comment]
I can't reproduce this on Linux; a screenshot of Crazy Castle is saved as `이상한 성 2018-05-24 11-55-31.png`. However, for unicode support, Windows uses a very different code path… Does this still happen for you?

**Comment by telk5093** [comment]
@AaronVanGeffen Yes.
When I (1)set in English, (2)open Crazy Castle, (3)set in Korean again, (4)take a screenshot,
then the scenario name is still ``Crazy Castle``, not ``이상한 성`` so that screenshot file is generated correctly.

If scenario name is written/displayed in korean, then file is not generated or incorrectly named for me.

**Comment by AaronVanGeffen** [comment]
This might be due to https://github.com/OpenRCT2/OpenRCT2/pull/7502 refactoring the screenshot routines, then. Could you test whether the problem **does** occur on 3278fd55dc3efee95daf2651350c32e738828fae ([build](https://openrct2.org/downloads/develop/3278fd55dc3efee95daf2651350c32e738828fae)), but **not** on 00278c6ca67d1f9c5af5e87b8f18a562d5ece94d ([build](https://openrct2.org/downloads/develop/00278c6ca67d1f9c5af5e87b8f18a562d5ece94d))?

**Comment by AaronVanGeffen** [comment]
With some help from a friend, I can confirm this was introduced to develop in 3278fd55dc3efee95daf2651350c32e738828fae ([build](https://openrct2.org/downloads/develop/3278fd55dc3efee95daf2651350c32e738828fae)). The commit right before the shared_ptr mergers (d284bcc4fe54162edbfd2fc664b08825e3450c1e, [build](https://openrct2.org/downloads/develop/d284bcc4fe54162edbfd2fc664b08825e3450c1e)) saves screenshots with the correct filenames.

Unfortunately, the commit right before this, (00278c6ca67d1f9c5af5e87b8f18a562d5ece94d, [build](https://openrct2.org/downloads/develop/00278c6ca67d1f9c5af5e87b8f18a562d5ece94d)) crashes while building the scenario list.

In short, this issue appears to have arisen in #7502, as I suspected. @IntelOrca, could you have a look?
